web/impact/impact/v1/views/cancel_office_hour_reservation_view.py

```diff
@@ -0,0 +1,91 @@
+from pytz import timezone
+
+from django.contrib.auth import get_user_model
+from django.template import loader
+
+from rest_framework.response import Response
+
+from impact.minimal_email_handler import MinimalEmailHandler as email_handler
+from impact.v1.views.impact_view import ImpactView
+
+from accelerator_abstract.models.base_user_utils import is_employee
+from accelerator.models import MentorProgramOfficeHour
+
+User = get_user_model()
+
+mentor_template_name = "cancel_office_hour_reservation_email_to_mentor.html"
+finalist_template_name = ("cancel_office_hour_reservation_email_to_finalist."
+                          "html")
+SUBJECT_LINE = "MassChallenge | Cancelled Office Hours with {} {}"
+
+
+class CancelOfficeHourReservationView(ImpactView):
+    view_name = "cancel_office_hours_reservation"
+
+    def post(self, request):
+        office_hour_id = request.POST.get("office_hour_id")
+        message = request.POST.get("message", "")
+        requesting_user_id = request.POST.get("user_id")
```

Comment:
We don't expect `requesting_user_id` to be different from `request.user.id`. This would be redundant.

Reply:
This is necessary since the staff can also cancel reservations

Reply:
Either way. When a staff user makes the api call, the still the `_can_cancel` check.